python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1380 @@
+      or amplifier-by-amplifier, depending on config.level.
+
+    - Once the cross-correlations have been calculated for each visit pair,
+      these are used to generate the correction kernel.
+      The maximum lag used, in pixels, and hence the size of the half-size
+      of the kernel generated, is given by config.maxLag,
+      i.e. a value of 10 will result in a kernel of size 2n-1 = 19x19 pixels.
+      Outlier values in these cross-correlations are rejected by using a
+      pixel-wise sigma-clipped thresholding to each cross-correlation in
+      the visitPair-length stack of cross-correlations.
+      The number of sigma clipped to is set by config.nSigmaClipKernelGen.
+
+    - Once DM-15277 has been completed, a method will exist to calculate the
+      empirical correction factor, config.biasCorr.
+      TODO: DM-15277 update this part of the docstring once the ticket is done.
+    """
+
+    RunnerClass = MakeBrighterFatterKernelTaskRunner
+    ConfigClass = MakeBrighterFatterKernelTaskConfig
+    _DefaultName = "makeBrighterFatterKernel"
+
+    def __init__(self, *args, **kwargs):
+        """Constructor for the MakeBrighterFatterKernelTask."""
+        pipeBase.CmdLineTask.__init__(self, *args, **kwargs)
+        self.makeSubtask("isr")
+
+        self.debug = lsstDebug.Info(__name__)
+        if self.debug.enabled:
+            self.log.info("Running with debug enabled...")
+            # if we're displaying, test it works and save the displays for later
+            # it's worth testing here as displays are flaky and sometimes
+            # can't be contacted, and given processing takes a while
+            # it's a shame to fail late due to display issues
+            if self.debug.display:
+                try:
+                    afwDisp.setDefaultBackend(self.debug.displayBackend)
+                    afwDisp.Display.delAllDisplays()
+                    self.disp1 = afwDisp.Display(0, open=True)
+                    self.disp2 = afwDisp.Display(1, open=True)
+
+                    im = afwImage.ImageF(1, 1)
+                    im.array[:] = [[1]]
+                    self.disp1.mtv(im)
+                    self.disp1.erase()
+                except NameError:
+                    self.debug.display = False
+                    self.log.warn('Failed to setup/connect to display! Debug display has been disabled')
+
+        plt.interactive(False)  # stop windows popping up when plotting. When headless, use 'agg' backend too
+        self.checkIsrConfig()
+        self.config.validate()
+        self.config.freeze()
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        """Augment argument parser for the MakeBrighterFatterKernelTask."""
+        parser = pipeBase.ArgumentParser(name=cls._DefaultName)
+        parser.add_argument("--visitPairs", nargs="*",
+                            help="Visit pairs to use. Each pair must be of the form INT,INT e.g. 123,456")
```

Comment:
Sorry I didn't catch this earlier, but all our existing command line options are lowercase with hyphens between words so please do that for the visit pairs argument. Use the dest keyword to specify the name for the parsed data, e.g.: `add_argument("--visit-pairs", dest="visitPairs", nargs="*", ...)`. This change will require a one or two tweaks to existing task documentation.